### ISSUE TITLE
Memory pointer ownership doesn't get transferred

### ISSUE THREAD

**Issue by skoch9** [fwilliams/numpyeigen#41]
Numpy doesn't own the data when arrays get returned from Eigen. This can probably also cause a memory leak...

**Comment by fwilliams** [comment]
This is not a bug. When NumpyEigen returns an array allocated in C++, it sets the [`base`](https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.ndarray.base.html) to be a [`PyCapsule`](https://docs.python.org/3.7/c-api/capsule.html) whose destructor callbacks frees the memory for that array. Thus, the owner of the array is the `base` capsule since it is responsible for freeing it. 

Basically, all NumPy arrays have an optional base object which is a reference to the owner of that array. a `PyCapsule` is a special `PyObject` which has a callback which is executed on object destruction.
